pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -756,6 +753,189 @@ template <class Precision> class StateVectorKokkos {
+     */
+
+    auto probs(const std::vector<size_t> &wires, const bool sorted_or_not) {
+        // Determining probabilities for the sorted wires.
+        const Kokkos::View<Kokkos::complex<Precision> *> arr_data = getData();
+        const size_t num_qubits = getNumQubits();
+
+        using UnmanagedSizeTHostView =
+            Kokkos::View<size_t *, Kokkos::HostSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+        using UnmanagedPrecisionHostView =
+            Kokkos::View<Precision *, Kokkos::HostSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+        using MDPolicyType_2D =
+            Kokkos::MDRangePolicy<Kokkos::Rank<2, Kokkos::Iterate::Left>>;
+
+        if (sorted_or_not) {
+            std::vector<size_t> all_indices =
+                Util::generateBitsPatterns(wires, num_qubits);
+
+            std::vector<size_t> all_offsets = Util::generateBitsPatterns(
+                Util::getIndicesAfterExclusion(wires, num_qubits), num_qubits);
+
+            Kokkos::View<Precision *> d_probabilities("d_probabilities",
+                                                      all_indices.size());
+
+            Kokkos::View<size_t *> d_sorted_ind_wires("d_sorted_ind_wires",
+                                                      wires.size());
+            Kokkos::View<size_t *> d_all_indices("d_all_indices",
+                                                 all_indices.size());
+            Kokkos::View<size_t *> d_all_offsets("d_all_offsets",
+                                                 all_offsets.size());
+
+            Kokkos::deep_copy(
+                d_all_indices,
+                UnmanagedSizeTHostView(all_indices.data(), all_indices.size()));
+            Kokkos::deep_copy(
+                d_all_offsets,
+                UnmanagedSizeTHostView(all_offsets.data(), all_offsets.size()));
+            Kokkos::deep_copy(
+                d_sorted_ind_wires,
+                UnmanagedSizeTHostView(const_cast<size_t *>(wires.data()),
+                                       wires.size()));
+
+            const int num_all_indices =
+                all_indices.size(); // int is required by Kokkos::MDRangePolicy
+            const int num_all_offsets = all_offsets.size();
+
+            MDPolicyType_2D mdpolicy_2d0({{0, 0}},
+                                         {{num_all_indices, num_all_offsets}});
+
+            Kokkos::parallel_for(
+                "Set_Prob", mdpolicy_2d0,
+                getSubProbFunctor<Precision>(arr_data, d_probabilities,
+                                             d_all_indices, d_all_offsets));
+
+            std::vector<Precision> probabilities(all_indices.size(), 0);
+
+            Kokkos::deep_copy(UnmanagedPrecisionHostView(probabilities.data(),
+                                                         probabilities.size()),
+                              d_probabilities);
+            return probabilities;
+
```

Comment:
Are these banches only different expect for the sorting step? We can simplify them further if this is the case.